CMakeLists.txt

```
@@ -773,6 +773,18 @@ endif()
 # MLIR/LLVM Dependency
 #-------------------------------------------------------------------------------
 
+# Both the IREE and MLIR Python bindings require nanobind. We initialize it here
+# at the top level so that everything uses ours consistently.
+if(IREE_BUILD_PYTHON_BINDINGS OR IREE_BUILD_COMPILER)
+  include(FetchContent)
+  FetchContent_Declare(
+    nanobind
+    GIT_REPOSITORY https://github.com/wjakob/nanobind.git
+    GIT_TAG        784efa2a0358a4dc5432c74f5685ee026e20f2b6 # 2.2.0
```

Depending on merge timing with https://github.com/iree-org/iree/pull/19421, update to 2.4.0 here

```suggestion
    GIT_TAG        0f9ce749b257fdfe701edb3cf6f7027ba029434a # v2.4.0
```

---

Well yeah, obviously I should have noticed this myself. Thanks for catching this Scott!

---

I rebase on top of the version bump.

---

Done.